### PR TITLE
Optional loadBalancerIP input in Contour

### PR DESCRIPTION
**Pull request by BostjanBozic**
Adds support for configuring Load Balancer IP in Envoy service. Previously, `loadBalancerIP` parameter of Envoy service was dynamically configured (from provider's side). Now user can preconfigure desired IP. This PR adds the following:
* An optional `loadBalancerIP` parameter to Contour's resource (`spec.networkPublish.envoy.loadBalancer.providerParameters`)

Updates https://github.com/projectcontour/contour-operator/issues/326

Signed-off-by: Bostjan Bozic <bozic.bostjann@gmail.com>

### CODE REVIEW

File: api/v1alpha1/contour_types.go [hunk 1]
```diff
@@ -266,6 +266,10 @@ type ProviderLoadBalancerParameters struct {
 	// +unionDiscriminator
 	// +kubebuilder:default=AWS
 	Type LoadBalancerProviderType `json:"type,omitempty"`
+
+	// loadBalancerIP contains IP for LoadBalancer service type, optional
+	// +optional
+	LoadBalancerIP string `json:"loadBalancerIP,omitempty"`
```
Comment (sunjayBhatia) [inline api/v1alpha1/contour_types.go:272]: would be good to add some validation that this string is in fact an ip address (here or in the validation package) so we can give feedback to users early if they have something wrong, and some min/max length validation on this string

File: internal/objects/service/service.go [hunk 2]
```diff
@@ -260,6 +260,9 @@ func DesiredEnvoyService(contour *operatorv1alpha1.Contour) *corev1.Service {
 	switch epType {
 	case operatorv1alpha1.LoadBalancerServicePublishingType:
 		svc.Spec.Type = corev1.ServiceTypeLoadBalancer
+		if len(contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.LoadBalancerIP) > 0 {
```
Comment (sunjayBhatia) [inline internal/objects/service/service.go:263]: would be good to have a unit test case for this

**Review by sunjayBhatia — CHANGES_REQUESTED** [review]

File: api/v1alpha1/contour_types.go (hunk 1 above)
Comment (BostjanBozic) [reply api/v1alpha1/contour_types.go:273]: Added `// +kubebuilder:validation:Format=ipv4` to have strict formating

**Comment by BostjanBozic** [comment]
I am sorry, I just made a mess here, I will create new PR.

File: internal/objects/service/service.go (hunk 2 above)
Comment (sunjayBhatia) [reply internal/objects/service/service.go:263]: i think that should be sufficient for now, its hard to do an end to end test for this b/c of kind + LoadBalancerService

**Pull request closed without merging**